DataStructure/src/main/list/ResizableArray.java

```diff
@@ -35,14 +43,19 @@ public void ensureCapacity(){
             list = list2;
         }
     }
+
     /**
      * Adds an element to the array.
      * @param num
      */
-    public void add(int num){
-        ensureCapacity();
-        list[writeIndex] = num;
-        writeIndex++;
+    public boolean add(int num){
+            ensureCapacity();
+            list[writeIndex] = num;
+            writeIndex++;
+            if (list[writeIndex - 1] == num){
+                return true;
+            }
+            return false;
```

Comment:
This line will never execute. There is no need of if statement. Line 53 will write the element at the index. The ensureCapacity will fail in some situations but those edge cases should be handled in ensureCapacity.